src/main/java/com/currencyexchange/cache/ExchangeRateCacheService.java

```diff
@@ -0,0 +1,56 @@
+package com.currencyexchange.cache;
+
+import com.currencyexchange.exception.RateNotFoundInCacheException;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class ExchangeRateCacheService {
+
+  private final Map<String, Map<String, Double>> exchangeRatesCache = new HashMap<>();
```

Comment:
BigDecimal instead of Double - https://www.baeldung.com/java-double-vs-bigdecimal